api/app/Http/Controllers/BaseController.php

```diff
@@ -1,39 +1,67 @@
 <?php namespace App\Http\Controllers;
 
+use App\Http\Transformers\BaseTransformer;
 use Laravel\Lumen\Routing\Controller;
 use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Illuminate\Http\Request;
+// use Illuminate\Container\Container as App;
 
 class BaseController extends Controller
 {
-
     public static $model;
 
+    /**
+     * Transformer for the response arrays.
+     *
+     * @var App\Services\Transformer
+     */
+    protected $transformer;
+
+    /**
+     * Model Repository.
+     *
+     * @var App\Repositories\BaseRepository
+     */
+    protected $repository;
+
+    /**
+     * Validation service.
+     *
+     * @var App\Services\Validation\TestValidator
+     */
+    protected $validator;
+
+    /**
+     * Get all entities.
+     *
+     * @return Response
+     */
     public function getAll()
     {
-        $model = static::$model;
-
-        $entities = $model::all();
-
-        return response()->json($entities);
-
-//        $this->transformer = $entity.'Transformer';
-//        return $this->response->collection($entities, new $this->transformer);
+        return $this->transformer->collection($this->repository->all(), new BaseTransformer);
```

Comment:
@xiphiaz I'm not completely sure I follow along here what you proposing.

So let's see, currently we inject the Transformer service as an dependency via the constructor, so any method can use it, that's the one that ends up as `$this->transformer`. That is the Services\Transformer class that exposes league\fractal to the application. Then the `BaseTransformer` class is the class to feed to `league\fractal` via the service, to define the actual transformation rules. So the basetransformer is the one that will be used in most cases, unless we need to extend it for any special situation. That one does not need to be injected.
 
I guess we could instead of injecting the transformer service in each controller, create it in the base controller constructor instead, and save some repeating code in the constructors. We cant inject it in a parent constructor, so we would then have to build it with a app->make() in the basecontroller, as Laravel can't do dependency injection in class hiearchies. I did that initially until I started looking into the testing part, ie had constructor hierarichy and used app->make() for the basecontroller. Is that what you are hinting at now, defining the service in the BaseController instead?

Or another option could be to make it to a full registered service, and add a Facade to it, that would remove the need to create it in controllers, and have it accessible at all time, and then just feed the BaseTransformer or an extended version to the "transformer" facade. 

What do you think?